iep-module-dynconfig/src/main/java/com/netflix/iep/dynconfig/DynamicConfigService.java

```diff
@@ -0,0 +1,142 @@
+
+  private final AtomicLong lastUpdateTime;
+  private final URI uri;
+  private final long pollingInterval;
+  private final boolean syncInit;
+  private final ScheduledExecutorService executor;
+
+  @Inject
+  DynamicConfigService(Registry registry) {
+    this.lastUpdateTime = PolledMeter.using(registry)
+        .withName("iep.archaius.cacheAge")
+        .monitorValue(
+          new AtomicLong(System.currentTimeMillis()),
+          Functions.AGE);
+
+    Config config = ConfigManager.get();
+    this.uri = URI.create(config.getString("netflix.iep.archaius.url"));
+    this.pollingInterval = config.getDuration("netflix.iep.archaius.polling-interval", TimeUnit.MILLISECONDS);
+    this.syncInit = config.getBoolean("netflix.iep.archaius.sync-init");
+    this.executor = Executors.newSingleThreadScheduledExecutor(r -> {
+      Thread t = new Thread(r, "DynamicConfigService");
+      t.setDaemon(true);
+      return t;
+    });
+  }
+
+  @Override protected void startImpl() throws Exception {
+    // Wait until properties have been updated at least once
+    while (syncInit) {
+      if (update()) {
+        break;
+      } else {
+        Thread.sleep(pollingInterval);
+      }
+    }
+
+    // Schedule for regular updates
+    executor.scheduleWithFixedDelay(this::update, pollingInterval, pollingInterval, TimeUnit.MILLISECONDS);
+  }
+
+  @Override protected void stopImpl() throws Exception {
+    executor.shutdown();
+    executor.awaitTermination(1, TimeUnit.MINUTES);
+  }
+
+  /**
+   * Query remote service and update the properties. Returns true if successful.
+   */
+  boolean update() {
+    try {
+      LOGGER.debug("updating properties from {}", uri);
+
+      HttpResponse response = HttpClient.DEFAULT_CLIENT.get(uri).send();
+
+      if (response.status() == 200) {
+        try (InputStream in = new ByteArrayInputStream(response.entity())) {
+          final Properties props = new Properties();
+          props.load(in);
+
+          if (LOGGER.isTraceEnabled()) {
+            props.stringPropertyNames().forEach(
+                k -> LOGGER.trace("received property: [{}] = [{}]", k, props.getProperty(k)));
+          }
+
+          updateDynamicConfig(props);
+          lastUpdateTime.set(System.currentTimeMillis());
+        }
+      } else {
+        throw new IOException("request failed with status: " + response.status());
+      }
+
+      return true;
+    } catch (Exception e) {
+      LOGGER.warn("failed to update dynamic properties", e);
+      return false;
+    }
+  }
+
+  /**
+   * Update the {@link ConfigManager#dynamicConfigManager()} with the properties. The value
+   * for a special key {@code netflix.iep.override} will be treated as a Typesafe Config string
+   * so that all constructs can be supported. Other properties will get used directly.
+   */
+  void updateDynamicConfig(Properties props) {
+    final String overrideKey = "netflix.iep.override";
+    if (props.containsKey(overrideKey)) {
```

Comment:
This is a clever combination of capabilities here - it's worth noting in the README, as it serves as the user guide. I would not have realized this without reading the source.

Reply:
Users shouldn't need to worry about it in most cases. I was planning to add an option to the deploy DSL that would handle this and would be the main way it is used.